src/CSLOLUtils.cpp

```diff
@@ -119,8 +119,79 @@ bool CSLOLUtils::isUnnecessaryAdmin() {
     return result;
 }
 #else
-// TODO: macos implementation
-QString CSLOLUtils::detectGamePath() { return ""; }
+#include <iostream>
+#include <filesystem>
+#include <string.h>
+#include <libproc.h>
+
+QString CSLOLUtils::detectGamePath() {
+    pid_t pid_list[2048];
+    int bytes = proc_listallpids(
+        pid_list, 
+        sizeof(pid_list));
```

Comment:
Necessary size can be queried by passing nullptr and 0 for size.
Quick google search shows an example of this in chromium:

https://chromium.googlesource.com/chromium/src/+/87ee38fb42b57edae0754e6af752a9ee9bc82843/chrome/browser/first_run/upgrade_util_mac.mm#61

It uses std::vector which would be QVector in Qt but if you are not familiar with that i'm willing to settle even for a malloc.

Reply:
I didn't actually know that, since I wasn't able to find any proper documentation on this. I just looked at quick source code snippets. But I'll change that, thank you. :+1: